qiskit_experiments/framework/analysis_result_table.py

```diff
@@ -10,256 +10,317 @@
+        key: str | int,
+    ) -> list[str]:
+        """Delete matched entries from this dataset.
 
         Args:
-            index: Name of entry to drop.
+            key: Identifier of the entry of interest.
 
-        Raises:
-            ValueError: When index is not in this table.
+        Returns:
+            Deleted analysis result IDs.
         """
+        uids = self._resolve_key(key)
         with self._lock:
-            if index not in self._container.index:
-                raise ValueError(f"Table index {index} doesn't exist in this table.")
-            self._container.drop(index, inplace=True)
+            self._data.drop(uids, inplace=True)
+
+        return uids
 
     def clear(self):
-        """Remove all elements from this container."""
+        """Clear all table entries."""
         with self._lock:
-            self._container = AnalysisResultContainer()
+            self._data = pd.DataFrame(columns=self.DEFAULT_COLUMNS)
+
+    def copy(self):
+        """Create new thread-safe instance with the same data.
 
-    def _unique_table_index(self):
-        """Generate unique UUID which is unique in the table with first 8 characters."""
+        .. note::
+            This returns a new object with shallow copied data frame.
+        """
+        with self._lock:
+            # Hold the lock so that no data can be added
+            new_instance = self.__class__()
+            new_instance._data = self._data.copy(deep=False)
+        return new_instance
+
+    def _create_unique_hash(self) -> str:
         with self._lock:
             n = 0
             while n < 1000:
                 tmp_id = str(uuid.uuid4())
-                if tmp_id[:8] not in self._container.index:
+                if tmp_id[:8] not in self._data.index:
                     return tmp_id
         raise RuntimeError(
             "Unique result_id string cannot be prepared for this table within 1000 trials. "
             "Reduce number of entries, or manually provide a unique result_id."
         )
 
-    def _repr_html_(self) -> Union[str, None]:
-        """Return HTML representation of this dataframe."""
+    def _resolve_columns(self, columns: str | list[str]):
         with self._lock:
-            return self._container._repr_html_()
+            extra_columns = [c for c in self._data.columns if c not in self.DEFAULT_COLUMNS]
+            if columns == "default":
+                return [
+                    "name",
+                    "experiment",
+                    "components",
+                    "value",
+                    "quality",
+                    "backend",
+                    "run_time",
+                ] + extra_columns
+            if columns == "minimal":
+                return [
+                    "name",
+                    "components",
+                    "value",
+                    "quality",
+                ] + extra_columns
+            if not isinstance(columns, str):
+                out = []
```

Comment:
I ran the tests with coverage to see how well tested ScatterTable and AnalysisResultTable were. Besides handling errors, this block for `get_data` with a list of column names was the only significant case not covered.

Reply:
Thanks for the test coverage check. Added more tests in [ee92f1d](https://github.com/Qiskit-Extensions/qiskit-experiments/pull/1360/commits/ee92f1df501db02c34505b52581d558cee964bae)